cpp/src/binaryop/compiled/Less.cu

```diff
@@ -14,13 +14,19 @@
  * limitations under the License.
  */
 
-#include "binary_ops.cuh"
+#include <binaryop/compiled/binary_ops.cuh>
+#include <binaryop/compiled/struct_binary_ops.cuh>
 
 namespace cudf::binops::compiled {
-template void apply_binary_op<ops::Less>(mutable_column_device_view&,
-                                         column_device_view const&,
-                                         column_device_view const&,
-                                         bool is_lhs_scalar,
-                                         bool is_rhs_scalar,
-                                         rmm::cuda_stream_view);
+template <>
+void apply_binary_op<ops::Less>(mutable_column_view& out,
+                                column_view const& lhs,
+                                column_view const& rhs,
+                                bool is_lhs_scalar,
+                                bool is_rhs_scalar,
+                                rmm::cuda_stream_view stream)
+{
+  detail::apply_binary_op_impl<ops::Less>(
```

Comment:
I see the problem of why this dispatch still feels so awkward. `apply_binary_op_impl` doesn't need to exist. Just put the logic of `apply_binary_op_impl` inside of `apply_binary_op`.

Reply:
That creates its other problems. To (slightly) resolve the explicit supported op comparisons, we're relying on the default implementation `apply_binary_op` not supporting structs. By putting the `apply_binary_op_impl` inside of `apply_binary_op` we'll lose that default.

What solution (explicit comparison lists for supported ops or awkward splitting of `apply_binary_op`) the lesser of the two evils?

Reply:
Reverted to explicitly maintained list of supported functions and merged `apply_binary_op_impl` and `apply_binary_op` per outside conversation.